### ISSUE TITLE
eosio-cpp compiles in a directory different than `.`

### ISSUE THREAD

**Issue by GassaFM** [EOSIO/eosio.cdt#535]
In my setup, eosio-cpp 1.6 couldn't find an include located in the same directory. So it apparently compiles stuff in a different directory. I solved it by explicitly adding `-I=$PWD` to the compiler options, but still wonder.

Is it a bug or a feature, or maybe my setup is flawed?

-----

Minimal example:
test.cpp:
~~~~~
#include "test.h"

using namespace eosio;

CONTRACT test : public eosio::contract
{
public:
	using contract::contract;

	ACTION testaction ()
	{
		require_auth (_self);
	}
};

EOSIO_DISPATCH (test, (testaction))
~~~~~

test.h:
~~~~~
#include <eosiolib/eosio.hpp>
~~~~~

Command line:
~~~~~
eosio-cpp test.cpp
~~~~~

Compiler output:
~~~~~
Warning, empty ricardian clause file
Warning, empty ricardian clause file
Warning, action <testaction> does not have a ricardian contract
/tmp/test.cpp:1:10: fatal error: './test.h' file not found
#include "./test.h"
         ^~~~~~~~~~
1 error generated.
/usr/local/eosio.cdt/bin/wasm-ld: error: cannot open /tmp/test.cpp.o: No such file or directory
/usr/local/eosio.cdt/bin/wasm-ld: error: cannot open /tmp/test.cpp.o: No such file or directory
~~~~~

So, looks like some stage of compilation happens in `/tmp/`.

I'd say it is a regression, since it was fine in previous versions of `eosio.cpp`.
Also, it can be considered a bug because it violates the principle of least surprise for C++ developers, and thus harms onboarding.

**Comment by conr2d** [comment]
Related to #472.

**Comment by jeffreyssmith2nd** [comment]
Closing as this was resolved in 1.6.2 by #472.
